tls-core/src/prf/mod.rs

```diff
@@ -70,34 +103,64 @@ mod tests {
         // H(ms xor opad)
         let outer_hash_state = partial_sha256_digest(&opad);
 
-        let (message, master) = master.next(inner_hash_state.clone());
-        let (message, slave) = slave.next(outer_hash_state.clone(), message);
+        let message = master.ke_setup(inner_hash_state).unwrap();
+        slave.ke_setup(outer_hash_state).unwrap();
+        let message = slave.next(message).unwrap();
 
         // H((ms xor opad) || H((ms xor ipad) || seed))
-        let a1 = message.a1.clone();
+        let a1 = if let PRFMessage::SlaveKe1(m) = message {
+            m.a1
+        } else {
+            panic!("unable to destructure");
+        };
         assert_eq!(
             &a1,
             &hmac_sha256(&ms, &seed_ke(&client_random, &server_random))
         );
 
-        let (message, master) = master.next(message);
-        let (message, slave) = slave.next(message);
+        let message = master.next(message).unwrap();
+        let message = slave.next(message).unwrap();
 
         // H((ms xor opad) || H((ms xor ipad) || a1))
-        let a2 = message.a2.clone();
+        let a2 = if let PRFMessage::SlaveKe2(m) = message {
+            m.a2
+        } else {
+            panic!("unable to destructure");
+        };
         assert_eq!(&a2, &hmac_sha256(&ms, &a1));
 
-        let (inner_hashes, master) = master.next(message);
+        let message = master.next(message).unwrap();
+        let (inner_hash_p1, inner_hash_p2) = if let PRFMessage::MasterKe3(m) = message {
+            (m.inner_hash_p1, m.inner_hash_p1)
+        } else {
+            panic!("unable to destructure");
+        };
 
-        let p1 = finalize_sha256_digest(outer_hash_state.clone(), 64, &inner_hashes.inner_hash_p1);
-        let p2 = finalize_sha256_digest(outer_hash_state, 64, &inner_hashes.inner_hash_p2);
+        let p1 = finalize_sha256_digest(outer_hash_state, 64, &inner_hash_p1);
+        let p2 = finalize_sha256_digest(outer_hash_state, 64, &inner_hash_p2);
 
         let mut ek = [0u8; 40];
         ek[..32].copy_from_slice(&p1);
         ek[32..].copy_from_slice(&p2[..8]);
+        let mut cwk = [0u8; 16];
+        cwk.copy_from_slice(&ek[..16]);
+
         let client_write_key = &ek[..16];
         let server_write_key = &ek[16..32];
         let client_write_iv = &ek[32..36];
         let server_write_iv = &ek[36..];
+
+        // Seems like the simplest way to compare byte slices is to hex encode
+        // them first.
+        assert_eq!(
+            hex::encode(&client_write_key),
+            "24542375e0909cabc9976eaa009d283a"
```

Comment:
Where did these static values come from?

Reply:
Those are the result on running this very test_prf() on the inputs. Should we pull in rustls as a dev dep and run the PRF with it to compare the output?

Reply:
I think instead of rustls Im gonna compare our output to this https://scapy.readthedocs.io/en/latest/api/scapy.layers.tls.crypto.prf.html

Reply:
As long as we check them against some other known working implementation we should be good. Either pulling in a dependency or pasting in is fine with me.